velox/common/memory/tests/MemoryManagerTest.cpp

```diff
@@ -387,6 +388,122 @@ TEST(MemoryManagerTest, concurrentPoolAccess) {
   pools.clear();
   ASSERT_EQ(manager.numPools(), 0);
 }
+
+TEST(MemoryManagerTest, quotaEnforcement) {
+  struct {
+    int64_t memoryQuotaBytes;
+    int64_t smallAllocationBytes;
+    int64_t largeAllocationBytes;
+    bool expectedMemoryExceedError;
+
+    std::string debugString() const {
+      return fmt::format(
+          "memoryQuotaBytes:{} smallAllocationBytes:{} largeAllocationBytes:{} expectedMemoryExceedError:{}",
+          succinctBytes(memoryQuotaBytes),
+          succinctBytes(smallAllocationBytes),
+          succinctBytes(largeAllocationBytes),
+          expectedMemoryExceedError);
+    }
+  } testSettings[] = {
+      {2 << 20, 1 << 20, 1 << 20, false},
+      {2 << 20, 1 << 20, 2 << 20, true},
+      {2 << 20, 2 << 20, 1 << 20, true},
+      {2 << 20, 3 << 20, 0, true},
+      {2 << 20, 0, 3 << 20, true}};
+
+  for (const auto& testData : testSettings) {
+    SCOPED_TRACE(testData.debugString());
+    std::vector<bool> contiguousAllocations;
```

Comment:
nit:
 
```
std::vector<bool> contiguousAllocations = {true, false};
```

Reply:
I see. Let me try this. I think some meta internal build tool will warn if I write like std::vector<bool> contiguousAllocations({false, true})